sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -573,4 +575,22 @@ public TableClientBuilder tableName(String tableName) {
 
         return this;
     }
+
+    /**
+     * Enabled tenant discovery when authenticating with the Storage Table Service. This is disabled by default.
+     * <p>
+     * Enable this if there is a chance for your application and the Storage account it communicates with to reside in
+     * different tenants. If this is enabled, clients created using this builder will make an unauthorized initial
+     * service request that will be met with a {@code 401} response containing an authentication challenge, which
+     * will be subsequently used to retrieve an access token to authorize all further requests with.
+     *
+     * @return The updated {@link TableClientBuilder}.
+     *
+     * @see TableBearerTokenChallengeAuthorizationPolicy
```

Comment:
This is not a public type and we should link to a non-public API as our JavaDocs will not include this file.